synapse/handlers/message.py

```diff
@@ -1013,16 +1029,22 @@ async def cache_joined_hosts_for_event(self, event: EventBase) -> None:
         # Note: We have to cache event ID -> prev state group, as we don't
         # store that in the DB.
         #
-        # Note: We always set the state group -> joined hosts cache, even if
-        # we already set it, so that the expiry time is reset.
+        # Note: We set the state group -> joined hosts cache if it hasn't been
+        # set for a while, so that the expiry time is reset.
 
         state_entry = await self.state.resolve_state_groups_for_events(
             event.room_id, event_ids=event.prev_event_ids()
         )
 
         if state_entry.state_group:
+            if self._external_cache_joined_hosts_updates:
```

Comment:
given that this whole method is conditional on `self._external_cache.is_enabled()`, can we just bung an `assert self._external_cache_joined_hosts_updates` up the top to save rechecking it twice?